### DISCUSSION TITLE
UI Not Popping Up and wg-client --wg-up Error: ModuleNotFoundError: No module named 'pynotify'

### DISCUSSION THREAD

**Discussion by demasj** [gene-git/wg-client#3] [General]
**Description:**

When attempting to start the UI, it does not appear. Additionally, running the command wg-client --wg-up results in the following error traceback:

```bash
Traceback (most recent call last):
  File "/usr/bin/wg-client", line 8, in <module>
    from lib import WgClient
  File "/usr/lib/python3.12/site-packages/wg_client/lib/__init__.py", line 7, in <module>
    from .class_client import WgClient
  File "/usr/lib/python3.12/site-packages/wg_client/lib/class_client.py", line 25, in <module>
    from .class_resolv import WgResolv
  File "/usr/lib/python3.12/site-packages/wg_client/lib/class_resolv.py", line 10, in <module>
    from pynotify import Inotify
ModuleNotFoundError: No module named 'pynotify'

```

**Environment:**

OS: Arch Linux  
KernelL 6.10.2-arch1-1
Python version: 3.12.4
wg-client version: 5.9.0

**Comment by gene-git** [comment]
Thanks for report - looks like a missing package which is a required dependency :

  - did you build this from the PKGBUILD (or aur)  or you running from git source?

While you can run it from git source it is much simpler to install the package. Its in the Arch  AUR for convenience as well. If you didn't build and install the package please try that first as it will take care of all the requirements to run.

In particular if you're trying to run from git source then please be sure all dependent packages are installed.
Logs for both gui and command line are saved into 

```
 ~/log/wg-client
 ~/log/wg-client-gui
```

From your trace it looks like at least the pynotify dependency is not installed (which  suggests it is being run from git source rather than installed).

Let's confirm you have the required dependencies - Can you please try running and see what is missing.

```
pacman --needed -Sy python-pyqt6  hicolor-icon-theme  python-psutil  python-netifaces  python-netaddr libcap  python-pynotify openssl
```

Also (see README for more details) using wireguard itself requires root privs and and for us that means the command line program requires that the  non-root user  be added to sudoers file. And note that the code also utilizes a couple of linux CAPABILITIES for a support tool (to manage dns resolv.conf ) which are set by the package installer.

thanks.

**Comment by gene-git** [comment]
By the way - python-pynotify is available in the [AUR](https://aur.archlinux.org/packages/wg-client) and source  is available on github [pynotify](https://github.com/gene-git/pynotify).

**Comment by gene-git** [comment]
Have you made any progress after installing package(s)?

Moving to discussion area since this looks like user installing issue rather than bug.

**Comment by githubkuyaya** [comment]
Just wanted to bump in and say I encountered a similar issue. The error on my side was `ImportError: cannot import name 'Inotify' from 'pynotify'`. Maybe there already was a `pynotify` package and it didn't overwrite it...? I don't know. To fix it, I uninstalled `wg-client-gui` again and tried to remove `pynotify` with pip, which gave me a permission error. Thus I forced it with sudo and --break-system-packages. Then, I reinstalled `wg-client-gui` again and it worked.

**Comment by gene-git** [comment]
Thank you @githubkuyaya its a very good point and indeed pip for system installs can cause problems - thanks for bringing this up.

I  use pacman and avoid pip entirely for any system installs.
